docs/examples/spectrum/spectral_factor.py

```diff
@@ -0,0 +1,140 @@
+"""
+Spectral Mismatch Estimation
+============================
+Comparison of spectral factor calculation methods used to estimate the spectral
+mismatch factor, :math:`M`, from atmospheric variable inputs.
+"""
+
+# %%
+# Introduction
+# ------------
+# This example demonstrates how to use different `spectrum.spectral_factor`
+# models in pvlib-python to calculate the spectral mismatch factor, :math:`M`.
```

Comment:
What is the correct way to typeset this if I want to refer to the group of spectral factor models in the spectrum module of pvlib?
Should I write it out with a hyperlink to [this page](https://pvlib-python.readthedocs.io/en/v0.11.0/reference/effects_on_pv_system_output/spectrum.html) perhaps? i.e.
```suggestion
# This example demonstrates how to use different spectral factor models from
# the [spectrum module](https://pvlib-python.readthedocs.io/en/v0.11.0/reference/effects_on_pv_system_output/spectrum.html) of pvlib-python to calculate the spectral mismatch factor, :math:`M`.
```

Reply:
```suggestion
# This example demonstrates how to use different ``spectrum.spectral_factor_*``
# models in pvlib-python to calculate the spectral mismatch factor, :math:`M`.
```
Adds code formatting (two backticks enclosing) and the `_*` suffix. I think it makes that more obvious.

Regarding linking, let's do another thing that may be out of scope for this PR; you commented something along grouping of `spectrum.spectral_factor` in our GSoC group. My intermediate shot would be:
1. Modifying `spectrum.rst` to group the functions,
2. Optionally link to that toctree (I think you would need to add an hyperlink on top of that section, as the one I added in the example below. See [:ref: docs](https://www.sphinx-doc.org/en/master/usage/referencing.html#ref-role). You would then refernece that with ```:ref:`spectrum_spectral_mismatch` ```):

```rst
.. currentmodule:: pvlib

Spectrum
--------

Spectra standards and simulators
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

.. autosummary::
   :toctree: ../generated/

   spectrum.spectrl2
   spectrum.get_reference_spectra
   spectrum.get_example_spectral_response
   spectrum.get_am15g

.. _spectrum_spectral_mismatch::

Spectral mismatch
^^^^^^^^^^^^^^^^^

.. autosummary::
   :toctree: ../generated/

   spectrum.calc_spectral_mismatch_field
   spectrum.spectral_factor_caballero
   spectrum.spectral_factor_firstsolar
   spectrum.spectral_factor_sapm
   spectrum.spectral_factor_pvspec
   spectrum.spectral_factor_jrc

Miscellaneous
^^^^^^^^^^^^^

.. autosummary::
   :toctree: ../generated/

   spectrum.sr_to_qe
   spectrum.qe_to_sr

```